ADALiOS/ADALiOS/ADAuthenticationContext.m

```diff
@@ -2226,6 +2239,32 @@ - (BOOL) canUseBroker
 }
 
 
+- (BOOL) respondsToUrl:(NSString*)url
+{
+    BOOL schemeIsInPlist = NO; // find out if the sceme is in the plist file.
+    NSBundle* mainBundle = [NSBundle mainBundle];
+    NSArray* cfBundleURLTypes = [mainBundle objectForInfoDictionaryKey:@"CFBundleURLTypes"];
+    if ([cfBundleURLTypes isKindOfClass:[NSArray class]] && [cfBundleURLTypes lastObject]) {
```

Comment:
This chunk of code would be clearer as a separate function with early returns then with control blocks nested 5 deep